editor/plugins/script_editor_plugin.cpp

```diff
@@ -4057,21 +4058,17 @@ void ScriptEditor::_on_find_in_files_modified_files(const PackedStringArray &pat
 	_update_modified_scripts_for_external_editor();
 }
 
-void ScriptEditor::_set_zoom_factor(float p_zoom_factor) {
+void ScriptEditor::_set_script_zoom_factor(float p_zoom_factor) {
 	if (zoom_factor == p_zoom_factor) {
 		return;
 	}
+
 	zoom_factor = p_zoom_factor;
-	for (int i = 0; i < tab_container->get_tab_count(); i++) {
-		ScriptEditorBase *se = Object::cast_to<ScriptEditorBase>(tab_container->get_tab_control(i));
-		if (se) {
-			CodeTextEditor *cte = se->get_code_editor();
-			if (cte) {
-				if (zoom_factor != cte->get_zoom_factor()) {
-					cte->set_zoom_factor(zoom_factor);
-				}
-			}
-		}
+}
+
+void ScriptEditor::_update_code_editor_zoom_factor(CodeTextEditor *p_code_text_editor) {
+	if (p_code_text_editor) {
```

Comment:
```suggestion
	if (p_code_text_editor && p_code_text_editor->is_visible() && zoom_factor != p_code_text_editor->get_zoom_factor()) {
```
These can also check if it is visible, since `visibility_changed` also emits when hiding, but we only want it when shown.
We should also keep the check to make sure it doesn't get set if it doesn't need to.
Same for the one in the shader editor.

Reply:
Sweet, added it to both editors.